tests/operator/operator.go

```diff
@@ -2297,12 +2231,12 @@ spec:
 					}},
 				},
 			}
-			errMsg := "spec.infra.nodePlacement.tolerations.operator in body should be one of"
-			patchKvInfra(&incorrectWorkload, true, errMsg)
+			const errMsg = "spec.infra.nodePlacement.tolerations.operator in body should be one of"
+			patchKVComponentConfig(originalKv, &incorrectInfra, infraField, MatchError(ContainSubstring(errMsg)))
```

Comment:
I the suggested function too smart, doing to many thing. I would have preferred to see something like
```
err := patchKvInfra(&incorrectWorkload)
Expect(err).To(HaveOccured())
```

(I'm not even sure we should have this patch validation in a webhook, but that's a different issue)

Reply:
not sure it should be part of this PR.